### ISSUE TITLE
Firebase requires reauth for Google Workspace Accounts

### ISSUE THREAD

**Issue by RafaelZasas** [firebase/firebase-tools#6898]
### [REQUIRED] Environment info

**firebase-tools: v13.5.2**

**Platform: Ubuntu 23.10**

### [REQUIRED] Steps to reproduce

Sign in to firebase-tools with a google workspace account.

Use firebase-tools as normal.

Wait a couple hours, firebase-tools breaks:
`FirebaseCommandException: An error occured on the Firebase CLI when attempting to run a command.`
From the log:

```
[debug] [2024-03-21T09:54:19.248Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-03-21T09:54:19.248Z] > authorizing via signed-in user (admin@<my_domain>)
[debug] [2024-03-21T09:54:19.251Z] > refreshing access token with scopes: []
[debug] [2024-03-21T09:54:19.251Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-03-21T09:54:19.252Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-03-21T09:54:19.706Z] <<< [apiv2][status] POST https://www.googleapis.com/oauth2/v3/token 400
[debug] [2024-03-21T09:54:19.706Z] <<< [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-03-21T09:54:19.706Z] Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[debug] [2024-03-21T09:54:20.110Z] FirebaseError: Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth
```

Running `firebase login` yields: 
```
Already logged in as admin@<my_domain>
```

### [REQUIRED] Expected behavior
Since this issue is not present with my personal account, I assume there are some settings set by Google Admin Console. I have checked the `
Google Cloud session control` and the `App Access Control` and there is no way to mark firebase cli as a trusted application, or extend the validation time for the oath token.

I expect to be able to login once, and have a valid refresh token for at least the same length of time as my other google services, but would ideally not have to log in again since I do not have to with my personal account.

### [REQUIRED] Actual behavior
In order to get firebase-tools working again, I have to run `firebase-login --reauth` which although, on the surface doesn't seem like a big deal, but it gets annoying to do every single day, and sometimes even after a short lunch break too.

**Comment by joehan** [comment]
Hey @RafaelZasas - thanks for reporting this! I see two improvements we culd make here:
1 - We should throw a clearer error when using expired credentials that prompts you to run the reauth command.
2 - We should offer firebase CLI as a trusted app in Google Admin Console.

1 I'll take a crack at when I have some free time soon. 2 will likely be longer, since we'll need to escalate to Google Workspaces team.

**Comment by cmjordan42** [comment]
Yeah, this is pretty bad that Google and Google don't work together. Signing in with my Google Workspaces account also seems to have bound me to now ALWAYS login with an account under that Google Workspaces @domain - certainly not something I asked it to do - so it prevents me from logging in to one of my Firebase admin non-Workspace accounts in order to address this bug.

I can't imagine it's that foreign for Firebase developers to also use Google Workspaces... I assumed that I hadn't configured something on Workspaces properly and it was being overly restrictive until I found this issue reported.

Please escalate this to Google Workspaces if you haven't already.

**Comment by joehan** [comment]
Hey @cmjordan42 - could you expand a bit on:

> Signing in with my Google Workspaces account also seems to have bound me to now ALWAYS login with an account under that Google Workspaces @my.domain - certainly not something I asked it to do - so it prevents me from logging in to one of my Firebase admin non-Workspace accounts in order to work around this bug.

What does this actually look like for you? When you run 'firebase login', are you not able to login with @gmail.com accounts? Do you see a different login screen?

**Comment by cmjordan42** [comment]
Sure.

1. `firebase login --reauth` gives a URL to accounts.google.com oauth
2. The sign in page displays a prompt on the left side **Choose an account from myworkspacedomain.com**, despite it launching in Chrome that has a) multiple accounts authenticated with Google accounts; b) a Gmail account logged in to Chrome (not the account or domain in question)
3. The right side has only **me@myworkspacedomain.com** as a user selection option, with **Use another account** below it.
4. Attempting to _use another account_ yields a sign in page where it's prompting for me to enter my email but with a forced (immutable) **@myworkspacedomain.com** domain to the email

**Comment by jibongil** [comment]
1. Download firebase cli for windows
2. (If there is a folder) delete C:\Users\(User)\.cache\firebase
3. Change the download file file name to **firebase-win.exe** and move it to C:\ path.
4. After running Powershell, run the following command and perform authentication.
 `C:\> ./firebase-win.exe login:ci`

5. (**Important**!) Copy the `firebase-win.exe` executable file to the Flutter project path and **_change the file name_** to `firebase.exe `(*When running the `flutterfire configure` command, (perhaps) it searches for firebase.exe and if it is not recognized, the error below will occur. )
```
D:\flutter_project> flutterfire configure
⠙ Fetching available Firebase projects...
FirebaseCommandException: An error occured on the Firebase CLI when attempting to run a command.
COMMAND: firebase --version
ERROR: The FlutterFire CLI currently requires the official Firebase CLI to also be installed, see https://firebase.google.com/docs/cli#install_the_firebase_cli for how to install it.
```

6. Run flutterfire configure command to create firebase_options.dart file

**Comment by Bullfrog1234** [comment]
Why is firebase-tools not requesting and using a refresh token. The hour long token is a standard access token length for lots of applications (including firebase auth). So, when using `--reauth` you are effectively starting the consent process from scratch like you have never gotten a token.

This is the only app that i have had issues with Google Workspace and never had to do anything to our Google Workspace to make it work.

Or am i missing something about this?

**Comment by Padreco-lbr** [comment]
This problem is not only present on firebase-tools, but also on the Firebase console itself, it constantly loses the authentication, forcing the user to sign in again. Maybe it's a Google Workspace auth issue? Any suggestions on how to change the auth life span?
